execution_engine/src/core/runtime/mint_internal.rs

```diff
@@ -105,8 +105,14 @@ where
     R: StateReader<Key, StoredValue>,
     R::Error: Into<execution::Error>,
 {
-    fn record_transfer(&mut self, source: URef, target: URef, amount: U512) -> Result<(), Error> {
-        let result = Runtime::record_transfer(self, source, target, amount);
+    fn record_transfer(
+        &mut self,
+        source: URef,
+        target: URef,
+        amount: U512,
+        id: Option<u64>,
+    ) -> Result<(), Error> {
+        let result = Runtime::record_transfer(self, source, target, amount, id);
```

Comment:
I saw something similar in different context but I'm curious why not `self.record_transfer`? That should work if I'm not mistaken

Reply:
Unfortunately it would recursively call this method and result in a stack overflow!